core-java-modules/core-java-11-2/src/main/java/com/baeldung/httpclient/parameters/HttpClientParametersExample.java

```diff
@@ -0,0 +1,250 @@
+package com.baeldung.httpclient.parameters;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HttpClientParametersExample {
+
+    public static void main(String[] args) throws Exception {
+        // Request with Query Parameters
+        queryParametersRequest();
+
+        // Request without any body
+        noBodyRequest();
+
+        // Request Body read as a String
+        stringBodyRequest();
+        stringBodyWithCharsetRequest();
+        jsonDataStringBodyRequest();
+        formDataStringBodyRequest();
+
+        // Request Body read as a Byte Array
+        byteArrayBodyRequest();
+        byteArrayBodyWithSpecificBytesRequest();
+        byteArraysIteratorRequest();
+
+        // Request Body read as a File
+        fileBodyRequest();
+
+        // Request Body read as an Input Stream
+        inputStreamBodyRequest();
+    }
+
+    public static void queryParametersRequest() throws IOException, InterruptedException {
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
+            .version(HttpClient.Version.HTTP_2)
+            .uri(URI.create("https://postman-echo.com/get?param1=value1&param2=value2"))
+            .GET()
+            .build();
```

Comment:
The long lines with which are broken in different lines such as the one which uses fluent API must be indented by 2 spaces.